boa_examples/src/bin/jstypedarray.rs

```diff
@@ -41,6 +44,80 @@ fn main() -> JsResult<()> {
         JsValue::new(sum)
     );
 
+    let greter_than_10_predicate = FunctionObjectBuilder::new(
+        context.realm(),
+        NativeFunction::from_fn_ptr(|_this, args, _context| {
+            let element = args
+                .get(0)
+                .cloned()
+                .unwrap_or_default()
+                .as_number()
+                .expect("error at number conversion");
+            Ok(JsValue::Boolean(element > 10.0))
+        }),
+    )
+    .build();
+
+    assert_eq!(
+        array.find_index(greter_than_10_predicate, None, context),
+        Ok(Some(11))
+    );
+
+    let lower_than_200_predicate = FunctionObjectBuilder::new(
+        context.realm(),
+        NativeFunction::from_fn_ptr(|_this, args, _context| {
+            let element = args
+                .get(0)
+                .cloned()
+                .unwrap_or_default()
+                .as_number()
+                .expect("error at number conversion");
+            Ok(JsValue::Boolean(element < 200.0))
+        }),
+    )
+    .build();
+
+    assert_eq!(
+        array.find_last(lower_than_200_predicate.clone(), None, context),
+        Ok(JsValue::Integer(199))
+    );
+
+    let data: Vec<u8> = vec![90, 120, 150, 180, 210, 240];
+    let array = JsUint8Array::from_iter(data, context)?;
+
+    assert_eq!(
+        array.find_last_index(lower_than_200_predicate, None, context),
+        Ok(Some(3))
+    );
+
+    // forEach
+    let array = JsUint8Array::from_iter(vec![1, 2, 3, 4, 5], context)?;
+    let num_to_modify = Gc::new(GcRefCell::new(0u8));
+
+    let js_function = FunctionObjectBuilder::new(
+        context.realm(),
+        NativeFunction::from_copy_closure_with_captures(
+            |_, args, captures, inner_context| {
+                let element = args
+                    .get(0)
+                    .cloned()
+                    .unwrap_or_default()
+                    .to_uint8(inner_context)
+                    .expect("error at number conversion");
+
+                captures.borrow_mut().add_assign(element);
+                Ok(JsValue::Undefined)
+            },
+            Gc::clone(&num_to_modify),
+        ),
+    )
+    .build();
+
+    let _unused = array.foreach(js_function, None, context);
+
+    let borrow = *Gc::clone(&num_to_modify).borrow();
```

Comment:
I don't think you need to clone the number again, you can just directly call `borrow` and compare with `15u8`.